src/tools/config_gen.zig

```diff
@@ -774,28 +774,57 @@ fn writeMarkdownFromHtmlInternal(html: []const u8, single_line: bool, depth: u32
     try writeLine(html[index..], single_line, writer);
 }
 
-/// takes in a signature like this: `@intToEnum(comptime DestType: type, integer: anytype) DestType`
-/// and outputs its arguments: `comptime DestType: type`, `integer: anytype`
-fn extractArgumentsFromSignature(allocator: std.mem.Allocator, signature: []const u8) error{OutOfMemory}![][]const u8 {
-    var arguments: std.ArrayListUnmanaged([]const u8) = .empty;
-    defer arguments.deinit(allocator);
+const Parameter = struct {
+    signature: []const u8,
+    type: ?[]const u8,
+};
+
+/// takes in a signature (without name or leading parenthesis) like this:
+/// `comptime DestType: type, integer: anytype) DestType`
+/// and outputs its parameters and return type:
+/// `comptime DestType: type`, `integer: anytype`, `DestType`
+fn extractParametersAndReturnTypeFromSignature(allocator: std.mem.Allocator, signature: []const u8) error{OutOfMemory}!struct { []Parameter, []const u8 } {
+    var parameters: std.ArrayListUnmanaged(Parameter) = .empty;
+    defer parameters.deinit(allocator);
 
     var argument_start: usize = 0;
+    var type_start: ?usize = null;
     var index: usize = 0;
-    while (std.mem.indexOfAnyPos(u8, signature, index, ",()")) |token_index| {
+    while (std.mem.indexOfAnyPos(u8, signature, index, ",():")) |token_index| {
         if (signature[token_index] == '(') {
-            argument_start = index;
-            index = 1 + std.mem.indexOfScalarPos(u8, signature, token_index + 1, ')').?;
+            index = token_index + 1;
+            var paren_depth: usize = 1;
+            while (paren_depth > 0) : (index += 1) {
+                switch (signature[index]) {
+                    '(' => paren_depth += 1,
+                    ')' => paren_depth -= 1,
+                    else => {},
+                }
+            }
```

Comment:
This fixes an existing bug where the parameter for `@errorFromInt` has a missing parenthesis

```zig
@errorFromInt(value: std.meta.Int(.unsigned, @bitSizeOf(anyerror))) anyerror
```